de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/v1_9/joiner/JoinerNodeUtil.java

```diff
@@ -109,8 +110,38 @@ public static void addIdentifierToParameters(
   }
 
   /**
-   * This method sets the default values of a combined model. The values are taken from the simulation settings 
-   * of the individual models.
+   * A helper method to check if the parameters have a SUFFIX_FIRST or SUFFIX_SECOND which means in
+   * the case of false that the workflow needs upgrade.
+   * 
+   * @param params the parameters to be tested
+   * @return
+   */
+  public static boolean parametersNeedUpdate(List<Parameter> params) {
+    boolean needsUpdate = false;
+    if(params!=null && params.size()>0) {
+        String firstID = params.get(0).getId();
+        char lastChar = firstID.charAt(firstID.length()-1);
+        
+        for (Parameter param : params) {
+          String paramID = param.getId();
+          char currectLastChar = paramID.charAt(paramID.length()-1);
+          if(currectLastChar == lastChar) {
+            if (!(paramID.endsWith(JoinerNodeModel.SUFFIX_FIRST)
+                || (paramID.endsWith(JoinerNodeModel.SUFFIX_SECOND)))) {
+              needsUpdate = true;
+            }
+          }else {
```

Comment:
- I think the assumption is wrong, the first and second iteration will not change the `needsUpdate` to true because they don't end with `JoinerNodeModel.SUFFIX_FIRST, JoinerNodeModel.SUFFIX_SECOND`.
- The change to false before break is done to cover a cases like: `log10_D1, log10_D1, dog` and this is covered by unittest.
- by reviewing the code I improve it to return the method on finding the first parameter doesn't end with either suffixes.
- it's a good catch for that the iteration has to start from the second item. solved in commit